src/profiler.js

```diff
@@ -0,0 +1,61 @@
+// profiler.js
+
+var _ = require('underscore');
+
+/**
+ * This RDF Pipeline component profiler gathers metrics on the processing time for each
+ * component and records them in the specific metrics directory.  The sequence of execution
+ * is also logged if createLog is called to enable logging. 
+ *
+ * @param node the RDF pipeline component facade
+ */
+module.exports = function(node) { 
+
+    _.extend(node, 
+        {
+          profiler: {
+              metrics: {
+                  averageUpdateTime: 0,
+                  numberOfErrors: 0,
+                  numberOfEvents: 0,
+                  numberOfUpdates: 0,
+                  startTime: Date.now(),
+                  totalErrorTime: 0,
+                  totalProcessingTime: 0,
+                  totalUpdateTime: 0
+              },
+
+              eventTypes: {
+                  ONDATA_EVENT: 1,
+                  UPDATE_SUCCESS: 2,
+                  UPDATE_ERROR: 3
+              },
+
+              update: function(startTime, eventType) { 
+                   var time = Date.now() - startTime + 1; // ensure we get at least 1 ms
```

Comment:
I did this for the profiler tests, but that becomes very ugly in the framework tests.  

If you don't care, I will take it out.  I felt that 1 ms round-up was acceptable for more reliable testing, since there is some time being spent in there, but if you feel it is not, I'll remove it.